api/hypershift/v1beta1/hosted_controlplane.go

```diff
@@ -202,6 +206,26 @@ type HostedControlPlaneSpec struct {
 	Labels map[string]string `json:"labels,omitempty"`
 }
 
+// kubeAPICustomKubeconfig is a group that englobes an additional KAS server address and the customkubeconfig generated for the kube-apiserver.
+// +optional
+type KubeAPICustomKubeconfig struct {
+	// customKubeConfig specifies the name and key for the External Custom kubeconfig secret.
+	// When set, it triggers the generation of a secret with the specified name containing a kubeconfig within the `HostedCluster` namespace.
+	// This kubeconfig will also be referenced in the `HostedCluster.status` as `customkubeconfig` (remarking that in the HC status is lowercase meanwhile this one is camelCase to keep the consistency with the rest of the spec).
+	// If removed during day-2 operations, all related secrets and status references will also be deleted.
+	// +optional
+	CustomKubeConfig *KubeconfigSecretRef `json:"customKubeConfig,omitempty"`
+
+	// kubeAPICustomName specifies the external name for the Kube API Server service.
+	// This is used to configure the Kube API Server service to use a different name, so
+	// the DNS should be configured to resolve the external name to the Kube API Server service.
+	// +kubebuilder:validation:XValidation:rule=`self == "" || self.matches('^(?:(?:[a-zA-Z0-9-]+\\.)+[a-zA-Z]{2,}|[a-zA-Z0-9-]+)$')`,message="kubeAPICustomName must be a valid URL name (e.g., api.example.com)"
+	// +kubebuilder:validation:MaxLength=253
+	// +kubebuilder:example: "api.example.com"
+	// +optional
+	KubeAPICustomName string `json:"kubeAPICustomName,omitempty"`
```

Comment:
this is within customKubeconfig shoud no need to name it "custom" again, also this is a dns name, right?
I would name similar to
customKubeConfig {
 kasDNSName
 secretRef
}